modules/jt-examples-of-common-use-cases.adoc

```diff
@@ -0,0 +1,172 @@
+      optional: true
+  params:
+    - name: repo-url
+    - name: revision
+    - name: context-path
+  tasks:
+    - name: fetch-repo
+      taskRef:
+        name: git-clone
+      workspaces:
+        - name: output
+          workspace: shared-workspace
+      params:
+        - name: url
+          value: "$(params.repo-url)"
+        - name: subdirectory
+          value: ""
+        - name: deleteExisting
+          value: "true"
+        - name: revision
+          value: $(params.revision)
+    - name: mvn-build
+      taskRef:
+        name: maven
+      runAfter:
+        - fetch-repo
+      workspaces:
+        - name: source
+          workspace: shared-workspace
+        - name: maven-settings
+          workspace: maven-settings
+      params:
+        - name: CONTEXT_DIR
+          value: "$(params.context-path)"
+        - name: GOALS
+          value: ["-DskipTests", "clean", "compile"]
+    - name: mvn-tests
+      taskRef:
+        name: maven
+      runAfter:
+        - mvn-build
+      workspaces:
+        - name: source
+          workspace: shared-workspace
+        - name: maven-settings
+          workspace: maven-settings
+      params:
+        - name: CONTEXT_DIR
+          value: "$(params.context-path)"
+        - name: GOALS
+          value: ["test"]
+    - name: mvn-package
+      taskRef:
+        name: maven
+      runAfter:
+        - mvn-tests
+      workspaces:
+        - name: source
+          workspace: shared-workspace
+        - name: maven-settings
+          workspace: maven-settings
+      params:
+        - name: CONTEXT_DIR
+          value: "$(params.context-path)"
+        - name: GOALS
+          value: ["package"]
+    - name: create-image-and-deploy
+      taskRef:
+        name: openshift-client
+      runAfter:
+        - mvn-package
+      workspaces:
+        - name: manifest-dir
+          workspace: shared-workspace
+        - name: kubeconfig-dir
+          workspace: kubeconfig-dir
+      params:
+        - name: SCRIPT
+          value: |
+            cd "$(params.context-path)"
+            mkdir -p ./artifacts/deployments && cp ./target/*.war ./artifacts/deployments
+            oc new-project helloworldproject
+            oc project helloworldproject
+            oc process -f jboss-eap70-binary-build.json | oc create -f -
+            oc start-build eap-helloworld-app --from-dir=artifacts/
+            oc new-app jboss-eap70-deploy.json
+
+----
+
+== Extending the core capabilities of Jenkins and Tekton by using plugins
+Jenkins has the advantage of a very rich ecosystem of numerous plugins developed over the years by its extensive user base. You can search and browse the plugins in the link:https://plugins.jenkins.io/[Jenkins Plugin Index]. 
+
+The comparatively newer Tekton also has many tasks developed and contributed by the community and enterprise users. A publicly available catalogue of reusable Tekton tasks are available in the link:https://hub.tekton.dev/[Tekton Hub].
```

Comment:
```suggestion
The comparatively newer Tekton also has many tasks developed and contributed by the community and enterprise users. A publicly available catalog of reusable Tekton tasks are available in the link:https://hub.tekton.dev/[Tekton Hub].
```
Should it be catalog instead because we say it Tekton catalog : https://github.com/tektoncd/catalog

Reply:
Yeah, my bad :-( . In spite of writing American English professionally for so many years now, I still make these mistakes.